src/java/org/apache/cassandra/cql3/Operator.java

```diff
@@ -359,20 +723,43 @@ public boolean isIN()
     }
 
     /**
-     * Checks if this operator is CONTAINS operator.
-     * @return {@code true} if this operator is a CONTAINS operator, {@code false} otherwise.
+     * Reverse this operator.
+     * @return the reverse operator from this operator.
+     */
+    public Operator negate()
+    {
+        throw new UnsupportedOperationException(this + " does not support negation");
+    }
+
+    /**
+     * Some operators are not supported by the read path because we never fully implemented support for them.
+     * It is the case for {@code IS_NOT} and {@code !=}
+     * @return {@code true} for the operators supported by the read path, {@code false} otherwise.
+     */
+    protected boolean isSupportedByReadPath()
+    {
+        return true;
+    }
+
+    /**
+     * The "LIKE_" operators are not real CQL operators and are simply a hack that should be removed at some point.
+     * Therefore, we want to ignore them for some operations.
+     * @return {@code true} for the "LIKE_" operators
      */
-    public boolean isContains()
+    private boolean isLikeXxx()
     {
-        return this == CONTAINS;
+        return name().startsWith("LIKE_");
     }
 
     /**
-     * Checks if this operator is CONTAINS KEY operator.
-     * @return {@code true} if this operator is a CONTAINS operator, {@code false} otherwise.
+     * Returns the operators that require an index or filtering for the specified column kind
+     * @param columnKind the column kind
+     * @return the operators that require an index or filtering for the specified column kind
      */
-    public boolean isContainsKey()
+    public static List<Operator> operatorsRequiringFilteringOrIndexingFor(ColumnMetadata.Kind columnKind)
     {
-        return this == CONTAINS_KEY;
+        return Arrays.stream(values())
```

Comment:
I know you like them but reconsider using streams :-)

Reply:
This method is only called to create an exception so it is not in any hot path.

Reply:
But what about future uses?

Reply:
The question in this case is: should we use Stream or not. I am totally fine with not using them in the hot path which I would defined like part potentially called hundredth or thousand of time. It is nowhere the case in this patch. It is used in parts of the code were it is barely used or at most twice. I believe that Stream are often easier to read and their past performance might be totally different from their future one. They are used in multiple places of C*. I am not introducing something new. If you want to go to the path of no stream in the code base we should open a discussion on the dev list.

Reply:
Hard -1 on this argument. I couldn't disagree more to it. We know nothing, unless there is some working time machine I haven't heard of, about the future performance of streams. This basically introduces an unnecessary risk in the codebase that `for` doesn't.

The rest of the comment simply doesn't apply as it's nothing of what I said.

We'll just disagree on this one and I tend to favor authorship.

Reply:
Nice one. I like it.

Reply:
After looking at the use case once more which is simply to provide more details in case of an error. I feel that we should probably keep things simple. There is no real performance issue here.